server/openapi.go

```diff
@@ -111,6 +112,9 @@ func (s *Server) sqlCheckController(c echo.Context) error {
 	if err != nil {
 		return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Database %s is not support", databaseType))
 	}
+	if catalog == nil {
```

Comment:
we can use

if request.DatabaseName == "" || request.Host == "" && request.Port == "" {
} else {
}

thus advisorDBType is enclosed in the if block

Reply:
NVM